janusgraph-server/src/main/java/org/janusgraph/core/CfgJanusGraphFactory.java

```diff
@@ -0,0 +1,133 @@
+        } else {
+            if (jgm != null) {
+                log.warn("You should supply \"graph.graphname\" in your .properties file configuration if you are opening " +
+                         "a graph that has not already been opened at server start, i.e. it was " +
+                         "defined in your YAML file. This will ensure the graph is tracked by the JanusGraphManager, " +
+                         "which will enable autocommit and rollback functionality upon all gremlin script executions. " +
+                         "Note that JanusGraphFactory#open(String === shortcut notation) does not support consuming the property " +
+                         "\"graph.graphname\" so these graphs should be accessed dynamically by supplying a .properties file here " +
+                         "or by using the ConfiguredGraphFactory.");
+            }
+            if (jgm == null) {
+                return new StandardJanusGraph(new GraphDatabaseConfigurationBuilder().build(configuration), new CfgManagementLoggerFactory());
+            }
+            return new StandardJanusGraph(new GraphDatabaseConfigurationBuilder().build(configuration));
+        }
+    }
+
+    /**
+     *  Return a Set of graph names stored in the {@link JanusGraphManager}
+     *
+     *  @return Set&lt;String&gt;
+     */
+    public static Set<String> getGraphNames() {
+       final JanusGraphManager jgm = JanusGraphManagerUtility.getInstance();
+       Preconditions.checkNotNull(jgm, JANUS_GRAPH_MANAGER_EXPECTED_STATE_MSG);
+       return jgm.getGraphNames();
+    }
+
+    /**
+     * Removes {@link Graph} from {@link JanusGraphManager} graph reference tracker, if exists
+     * there.
+     *
+     * @param graph Graph
+     */
+    public static void close(Graph graph) throws Exception {
+        final JanusGraphManager jgm = JanusGraphManagerUtility.getInstance();
+        if (jgm != null) {
+            jgm.removeGraph(((StandardJanusGraph) graph).getGraphName());
+        }
+        graph.close();
+    }
+
+    /**
+     * Drop graph database, deleting all data in storage and indexing backends. Graph can be open or closed (will be
+     * closed as part of the drop operation). The graph is also removed from the {@link JanusGraphManager}
+     * graph reference tracker, if there.
+     *
+     * <p><b>WARNING: This is an irreversible operation that will delete all graph and index data.</b></p>
+     * @param graph JanusGraph graph database. Can be open or closed.
+     * @throws BackendException If an error occurs during deletion
+     */
+    public static void drop(JanusGraph graph) throws BackendException {
+        Preconditions.checkNotNull(graph);
+        Preconditions.checkArgument(graph instanceof StandardJanusGraph,"Invalid graph instance detected: %s",graph.getClass());
+        final StandardJanusGraph g = (StandardJanusGraph) graph;
```

Comment:
Same comment here as in `JanusGraphFactory` as https://github.com/JanusGraph/janusgraph/pull/2111#discussion_r429775308